charts/opentelemetry-demo/templates/otelcol.yaml

```diff
@@ -1,81 +0,0 @@
-{{- $ := set . "name" "otelcol" }}
-{{- $otelconfig := include "otel-demo.otelcol.config" .}}
-
-{{- if .Values.observability.otelcol.enabled -}}
----
-apiVersion: v1
-kind: ConfigMap
-metadata:
-  name: {{ include "otel-demo.name" . }}-{{ .name }}-config
-  labels:
-    {{- include "otel-demo.labels" . | nindent 4 }}
-data:
-  otelcol-config.yml: |
-
-    {{- $otelconfig | nindent 4 }}
----
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: {{ include "otel-demo.name" . }}-{{ .name }}
-  labels:
-    {{- include "otel-demo.labels" . | nindent 4 }}
-spec:
-  replicas: 1
-  selector:
-    matchLabels:
-      {{- include "otel-demo.selectorLabels" . | nindent 6 }}
-  template:
-    metadata:
-      labels:
-        checksum/otel_config: {{ sha256sum $otelconfig | trunc 63 }}
-        {{- include "otel-demo.selectorLabels" . | nindent 8 }}
-    spec:
-      containers:
-      - image: otel/opentelemetry-collector:0.55.0
-        name: otel-col
-        args: ["--config=/etc/otelcol-config.yml"]
-        {{- if .Values.observability.jaeger.enabled }}
-        env:
-          - name: JAEGER_ADDR
-            value: {{ include "otel-demo.name" . }}-jaeger:14250
```

Comment:
@dmitryax we need to be able to tell the subchart what the endpoint will be fore the jaeger exporter.  The demo chart currently creates the the name of the jaeger deployment based on the release name, but we can't get access to the release name in the values.yaml when setting the subchart's env vars or collector config.

How do you want to handle this?  Hardcode the jaeger name seems like a simple option, but feels like an anti-pattern.  What do you think?